pkg/server/server.go

```diff
@@ -167,6 +177,73 @@ func (s *server) GetApplication(ctx context.Context, msg *pb.GetApplicationReque
 	}}, nil
 }
 
+//Until the middleware is done this function will not be able to get the token and will fail
+func (s *applicationServer) ListCommits(ctx context.Context, msg *pb.ListCommitsRequest) (*pb.ListCommitsResponse, error) {
+	//Remove when middleware is done
+	type contextVals struct {
+		Token *oauth2.Token
+	}
+
+	vals := contextVals{Token: &oauth2.Token{AccessToken: "temptoken"}}
+	ctx = context.WithValue(ctx, tokenKey, vals)
+
+	//Temporary solution to get this to build until middleware is done
+	extractToken := func(ctx context.Context) (string, error) {
```

Comment:
Does this utilize anything from the outer scope of the handler? This is a closure pattern, but I don't see where it is closing over the outer scope variables. Can we move this out of the handler into a local function? You can move it right above this `ListCommits` handler if you want so its easily readable.

Same with the `contextVals` struct.